### ISSUE TITLE
SocketManager.init failing with config options

### ISSUE THREAD

**Issue by cbiggin** [socketio/socket.io-client-swift#1200]
The following code fails on v15.1.0:

`self.manager = SocketManager.init(socketURL: url, config: [.log(true), .compress])`

If you eliminate the config options, then it works. The above works under v15.0.0.

**Comment by nuclearace** [comment]
What does it fail with?

**Comment by cbiggin** [comment]
The following is working as well:

`self.manager = SocketManager.init(socketURL: url, config: [.compress])`

So it looks like it's the `.log` option.

**Comment by cbiggin** [comment]
![Screen Shot 2019-05-29 at 1 08 47 PM](https://user-images.githubusercontent.com/66640/58576740-01575180-8213-11e9-8046-cee4df9b307f.png)

**Comment by nuclearace** [comment]
That's a bit strange... and also a little unhelpful. Does it print anything? Are you running this is in a debug config or a release.

**Comment by nuclearace** [comment]
Also, have you done a fresh install/compile? I'd include a CocoaPods refresh as part of that, if CocoaPods is involved.

**Comment by cbiggin** [comment]
Nope, not printing anything. And yes, running it in debug config. Also failing on both simulator and device. Using Carthage.

Just created a small project and that seems to initialize fine so will delve deeper.

**Comment by nuclearace** [comment]
One thing to note, nothing in this library is thread safe. So if you had some concurrent access violation happening, that could cause it.

**Comment by sbachmannlp** [comment]
Hi, i was running into the same problem, my setup is identical to @cbiggin post above https://github.com/socketio/socket.io-client-swift/issues/1200#issuecomment-497036884 ; removing the `.log` option from `SocketIOClientConfiguration` seems to help for now

**Comment by abmomen** [comment]
Encountered same issue, with no log or anything. Xcode debug stops with EXC_BAD_ACCESS(code=1, address=0x0) at 
```SocketIO.SocketManager.init(socketURL: Foundation.URL, config: SocketIO.SocketIOClientConfiguration) -> SocketIO.SocketManager```

Xcode: 13.4.1
 Podfile: pod 'Socket.IO-Client-Swift', '~> 16.0.0'
